ico.c

```diff
@@ -1097,12 +1100,52 @@ do_ico_window(void *ptr)
 			icoX -= icodeltax2;
 			icoDeltaX = - icoDeltaX;
 			icodeltax2 = icoDeltaX * 2;
+
+			if (useDvd)
+			{
+				XColor cdef, igndef;
+
+				currentDvdColor++;
+				if (currentDvdColor < 0 || currentDvdColor > numcolors - 1)
+				{
+					currentDvdColor = 0;
+				}
+				if (XAllocNamedColor(dpy, closure->cmap, colornames[currentDvdColor], &cdef, &igndef))
+					fg = cdef.pixel;
+				else
+					icoFatal("face: no such color \"%s\"", colornames[currentDvdColor]);
+				xgcv.foreground = fg;
+				closure->gcontext = XCreateGC(dpy, closure->draw_window, vmask, &xgcv);
```

Comment:
There might be a possible memory leak when creating a new GC without free-ing the existing GC here. We'd investigate the actual run to see if it actually leaks memory.